test/node/symbol.test.ts

```diff
@@ -0,0 +1,44 @@
+import { expect } from 'chai';
+import { BSONSymbol, BSON } from '../register-bson';
+import { bufferFromHexArray } from './tools/utils';
+
+describe('class BSONSymbol', () => {
+  it('get _bsontype returns BSONSymbol', () => {
+    const sym = new BSONSymbol('symbol');
+    expect(sym).to.have.property('_bsontype', 'BSONSymbol');
+  });
+
+  it('serializes to a bson symbol type', () => {
+    const bytes = bufferFromHexArray([
+      '0E', // bson symbol
+      Buffer.from('sym\x00', 'utf8').toString('hex'),
+      '07000000', // 6 bytes
+      Buffer.from('symbol\x00').toString('hex')
+    ]);
+
+    expect(BSON.serialize({ sym: new BSONSymbol('symbol') })).to.deep.equal(bytes);
+  });
+
+  it('deserializes to js string by default', () => {
+    const bytes = bufferFromHexArray([
+      '0E', // bson symbol
+      Buffer.from('sym\x00', 'utf8').toString('hex'),
+      '07000000', // 6 bytes
+      Buffer.from('symbol\x00').toString('hex')
+    ]);
+
+    expect(BSON.deserialize(bytes)).to.have.property('sym', 'symbol');
+  });
+
+  it('deserializes to BSONSymbol if promoteValues is false', () => {
```

Comment:
why is `promoteValues` significant here?

Reply:
promoteValues controls whether a string or BSONSymbol instance is returned. I wrote these test b/c tests were failing in the driver but not here in BSON early on in this PR